pkg/intermediate/aggregate.go

```diff
@@ -59,18 +59,22 @@ func InitAggregationProcess(messageChan chan *entities.Message, workerNum int, c
 }
 
 func (a *AggregationProcess) Start() {
+	a.mutex.Lock()
 	for i := 0; i < a.workerNum; i++ {
 		w := createWorker(i, a.messageChan, a.AggregateMsgByFlowKey)
 		w.start()
 		a.workerList = append(a.workerList, w)
 	}
+	a.mutex.Unlock()
```

Comment:
is there an issue using defer unlock?

Reply:
Since the stopChan will keep waiting for stop signal, using defer will not release the lock until stopChan receives stop signal.